lib/auth/auth_test.go

```diff
@@ -685,3 +689,123 @@ func (s *AuthSuite) TestUpsertDeleteRole(c *C) {
 	c.Assert(deleteEventEmitted, Equals, false)
 
 }
+
+func (s *AuthSuite) TestGithubConnectorCRUDEventsEmitted(c *C) {
+	eventEmitted := false
+	s.mockedAuditLog.MockEmitAuditEvent = func(event events.Event, fields events.EventFields) error {
+		eventEmitted = true
+		c.Assert(event, DeepEquals, events.GithubConnectorCreated)
+		return nil
+	}
```

Comment:
This seems like a very common pattern.
Maybe `mockAuditLog` can make this easier?

For example: `mockAuditLog` can keep a slice of emitted events internally and the caller can do something like:
```
s.a.upsertGithubConnector(github)
c.Assert(s.mockAuditLog.EmittedEvents, check.DeepEquals, []events.Event{
  events.GithubConnectorCreated,
})
s.a.upsertGithubConnector(github)
c.Assert(s.mockAuditLog.EmittedEvents, check.DeepEquals, []events.Event{
  events.GithubConnectorCreated,
  events.GithubConnectorCreated,
})
```

Reply:
Thank you for the suggestion, I refactored all the tests: https://github.com/gravitational/teleport/pull/3736/commits/ce15f25bbdf631897d4753aefcd151c4b3ab8963

Reply:
I think I might need to refactor tests in another PR, i have some other PR's that are related to this...